addons/mail/tests/test_res_partner.py

```diff
@@ -90,6 +90,44 @@ def test_parse_partner_name(self):
             self.assertEqual(parsed[0], expected_name)
             self.assertEqual(parsed[1], expected_email)
 
+
+    def test_res_partner_address_tracking(self):
+        company_partner = self.env.company.partner_id
+        # use some wacky formatting to check inlining
+        company_partner.country_id.address_format = """%(street)s
+            \n\n\n%(street2)s
+            %(city)s %(state_code)s %(zip)s
+            \n%(country_name)s\n
+            \n   """
+        company_partner.write({
+            'city': 'Some City Name',
+            'street': 'Some Street Name',
+            'type': 'contact',
+        })
+        child_partner = self.env['res.partner'].create({
+            'name': 'Some Guy',
+            'parent_id': company_partner.id,
+        })
+        self.env.flush_all()
+        self.cr.precommit.run()
+        # keep track of setup messages
+        partners = (company_partner, child_partner)
+        partner_original_messages = (company_partner.message_ids, child_partner.message_ids)
+
+        company_partner.street = 'Some Other Street Name'
```

Comment:
Ah right, that was the whole point of setting it in the setup :facepalm: 

I'm doing it in a single write as I believe doing it in 2 steps would normally create 2 messages. Being 2 x (write + flush). Just so it's more intuitive.

Reply:
Yeah but in a real usecase it would be more like:
record.street = street1
flush()
record.street = street2
flush()

Right?

Ok with using 2 assignments if I'm wrong though